src/integrations/pulsar/consumer.cpp

```diff
@@ -0,0 +1,285 @@
+// Copyright 2021 Memgraph Ltd.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt; by using this file, you agree to be bound by the terms of the Business Source
+// License, and you may not use this file except in compliance with the Business Source License.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+#include "integrations/pulsar/consumer.hpp"
+
+#include <fmt/format.h>
+#include <pulsar/Client.h>
+
+#include <chrono>
+#include <thread>
+
+#include "integrations/constants.hpp"
+#include "integrations/pulsar/exceptions.hpp"
+#include "utils/concepts.hpp"
+#include "utils/logging.hpp"
+#include "utils/on_scope_exit.hpp"
+#include "utils/result.hpp"
+#include "utils/thread.hpp"
+
+namespace integrations::pulsar {
+
+namespace {
+
+template <typename T>
+concept PulsarConsumer = utils::SameAsAnyOf<T, pulsar_client::Consumer, pulsar_client::Reader>;
+
+template <PulsarConsumer TConsumer>
+utils::BasicResult<std::string, std::vector<Message>> GetBatch(TConsumer &consumer, const ConsumerInfo &info,
+                                                               std::atomic<bool> &is_running) {
+  std::vector<Message> batch{};
+
+  const auto batch_size = info.batch_size.value_or(kDefaultBatchSize);
+  batch.reserve(batch_size);
+
+  auto remaining_timeout_in_ms = info.batch_interval.value_or(kDefaultBatchInterval).count();
```

Comment:
Why not just `info.batch_interval.value_or(kDefaultBatchInterval)` (keep the chrono type). That way, later on you don't need to use `.count()`

Reply:
I compare the `remaining_timeout_in_ms` in the for loop to an integer so I would need to fetch count everytime.